core/src/main/java/org/apache/calcite/util/Pair.java

```diff
@@ -185,7 +185,8 @@ public Pair(T1 left, T2 right) {
     final int size;
     if (strict) {
       if (ks.size() != vs.size()) {
-        throw new AssertionError();
+        throw new IllegalArgumentException(
+            "The size of two lists to be combined is required to be equal.");
```

Comment:
size -> sizes
is -> are

Reply:
Should the sizes be included in the message as well?